packages/caver-wallet/src/index.js

```diff
@@ -16,13 +16,309 @@
+
+        founded.key = keyring.copy().key
+        return founded
+    }
+
+    /**
+     * querys the keyring address, and return.
+     *
+     * @param {string} address The address of keyring to query.
+     * @return {Keyring}
+     */
+    getKeyring(address) {
+        if (!utils.isAddress(address))
+            throw new Error(
+                `Invalid address ${address}. To get keyring from wallet, you need to pass a valid address string as a parameter.`
+            )
+
+        const founded = this._addressKeyringMap.get(address.toLowerCase())
+
+        return founded
+    }
+
+    /**
+     * adds a keyring to the keyringContainer.
+     *
+     * @param {Keyring} keyring A keyring instance to add to keyringContainer.
+     * @return {Keyring}
+     */
+    add(keyring) {
+        if (this._addressKeyringMap.get(keyring.address.toLowerCase()) !== undefined)
+            throw new Error(`Duplicate Account ${keyring.address}`)
+
+        const keyringToAdd = keyring.copy()
+
+        this._length++
+        this._addressKeyringMap.set(keyringToAdd.address.toLowerCase(), keyringToAdd)
+
+        return keyringToAdd
+    }
+
+    /**
+     * deletes the keyring that associates with the given address from keyringContainer.
+     *
+     * @param {string} address An address of the keyring to be deleted in keyringContainer.
+     * @return {boolean}
+     */
+    remove(address) {
+        let keyringToRemove
+        if (utils.isAddress(address)) {
+            keyringToRemove = this.getKeyring(address)
+        } else {
+            throw new Error(`To remove keyring, parameter should be address string.`)
+        }
+
+        if (keyringToRemove === undefined) return false
+
+        keyringToRemove.key = null
+        this._addressKeyringMap.delete(keyringToRemove.address.toLowerCase())
+
+        this._length--
+
+        return true
+    }
+
+    /**
+     * signs with data and returns the result object that includes `signature`, `message` and `messageHash`
+     *
+     * @param {string} address An address of keyring in keyringContainer.
+     * @param {string} data The data string to sign.
+     * @param {number} [role] A number indicating the role of the key. You can use `caver.wallet.keyring.role`.
+     * @param {number} [index] An index of key to use for signing.
+     * @return {object}
+     */
+    signMessage(address, data, role, index) {
+        const keyring = this.getKeyring(address)
+        if (keyring === undefined) throw new Error(`Failed to find keyring from wallet with ${address}`)
+        return keyring.signMessage(data, role, index)
+    }
+
+    /**
+     * signs the transaction using one key and return the transactionHash
+     *
+     * @param {string} address An address of keyring in keyringContainer.
+     * @param {Transaction} transaction A transaction object.
+     * @param {number} [index] An index of key to use for signing.
```

Comment:
@jimni1222 Is there any way to describe the optional parameters in the comment?

Reply:
I followed JSDoc format, and they use '[]' to describe optional parameter.
So in this case [index] parameter is optional.